ui_framework/components/button/kui_button.js

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import classnames from 'classnames';
+
+import { KuiButtonLink } from './kui_button_link';
+import { KuiTooltip } from '../tooltip/kui_tooltip';
+
+export function KuiButton(props) {
+  if (props.href && props.onClick) {
+    throw new Error('Only href or onClick can be defined, not both');
+  }
+
+  if (props.hide) return null;
+
+  const classes = classnames('kuiButton', props.className);
+
+  if (props.href) {
+    return <KuiButtonLink {...props} />;
+  } else {
+    return <KuiTooltip text={ props.tooltip }>
+      <button
+        className={ classes }
+        aria-label={ props.tooltip }
+        onClick={ props.onClick }
+      >
+        { props.children }
+      </button>
+    </KuiTooltip>;
```

Comment:
Another benefit of this approach is that it makes it easier to migrate from many Tooltip instances to using a single Tooltip instance in an owner component. For example, if we create a VisualizeListing component, we can add a Tooltip instance in that component's render method, and then set `tooltip` to false/undefined for all of the components it owns